airbyte-webapp/src/hooks/services/ConnectionEdit/ConnectionEditService.tsx

```diff
@@ -11,22 +17,69 @@ interface ConnectionEditProps {
   connectionId: string;
 }
 
-const useConnectionEdit = ({ connectionId }: ConnectionEditProps) => {
-  const [connection, setConnection] = useState(useGetConnection(connectionId));
+export interface ConnectionCatalog {
+  syncCatalog: AirbyteCatalog;
+  catalogId?: string;
+}
+
+interface ConnectionEditHook {
+  connection: WebBackendConnectionRead;
+  connectionUpdating: boolean;
+  schemaError?: Error;
+  schemaRefreshing: boolean;
+  schemaHasBeenRefreshed: boolean;
+  updateConnection: (connectionUpdates: WebBackendConnectionUpdate) => Promise<void>;
+  refreshSchema: () => Promise<void>;
+  discardRefreshedSchema: () => void;
+}
+
+const getConnectionCatalog = (connection: WebBackendConnectionRead): ConnectionCatalog =>
+  pick(connection, ["syncCatalog", "catalogId"]);
+
+const useConnectionEdit = ({ connectionId }: ConnectionEditProps): ConnectionEditHook => {
   const connectionService = useWebConnectionService();
+  const [connection, setConnection] = useState(useGetConnection(connectionId));
+  const [catalog, setCatalog] = useState<ConnectionCatalog>(() => getConnectionCatalog(connection));
```

Comment:
I feel like this could just be a memo? I guess I'm not wholly certain what holding these as a separate value gets us. Does holding it like this remove the issue w/ changing the name or toggling the connection resetting the form?

Reply:
The value of the catalog is controlled in case the schema update is canceled after the refresh, so it's only setting the catalog when the update has been committed and undoing when the update is canceled.